sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManager.scala

```diff
@@ -232,10 +231,21 @@ abstract class RapidsShuffleInternalManagerBase(conf: SparkConf, isDriver: Boole
 
   private lazy val localBlockManagerId = blockManager.blockManagerId
 
+  // Code that expects the shuffle catalog to be initialized gets it this way,
+  // with error checking in case we are in a bad state.
+  private def getCatalogOrThrow: ShuffleBufferCatalog = {
+    val catalog = GpuShuffleEnv.getCatalog
+    if (catalog == null) {
+      throw new IllegalStateException("The ShuffleBufferCatalog is not initialized but the " +
+        "RapidsShuffleManager is configured")
+    }
+    catalog
```

Comment:
nit: we can abbreviate this code with this kind of idiom:
```
Option(GpuShuffleEnv.getCatalog).getOrElse(
  throw new IllegalStateException("The ShuffleBufferCatalog is not initialized but the " +
        "RapidsShuffleManager is configured"
)
```

Reply:
I'll get this changed shortly.